unlock-app/src/components/interface/checkout/main/useCheckoutHook.ts

```diff
@@ -0,0 +1,95 @@
+import { useQueries } from '@tanstack/react-query'
+import { networks } from '@unlock-protocol/networks'
+import { HookType } from '@unlock-protocol/types'
+import { useActor } from '@xstate/react'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+import { CheckoutHookType, CheckoutService } from './checkoutMachine'
+
+type LockHookProps = Record<string, CheckoutHookType | undefined>
+const HookIdMapping: Partial<Record<HookType, CheckoutHookType>> = {
+  PASSWORD: 'password',
+}
+
+export function useCheckoutHook(service: CheckoutService) {
+  const [state] = useActor(service)
+  const web3Service = useWeb3Service()
+
+  let lockHookMapping: LockHookProps = {}
+  const { paywallConfig } = state.context
+
+  const queries = useQueries({
+    queries: [
+      ...Object.entries(paywallConfig.locks).map(
+        ([lockAddress, { network = 1 }]) => {
```

Comment:
hm this way, we can't batch the request. Do you think it would make sense to use Promise.all under a single query?

Reply:
Actually the `Promise.all` it what i had before this comment https://github.com/unlock-protocol/unlock/pull/11288#discussion_r1130865210 ? 

wdyt?

Reply:
Merging for now! ill change it later eventually

Reply:
I think you got confused. What I meant was separate the request code into a separate promise.all call.